### ISSUE TITLE
interactive_builder.py Fix or upgrade?

### ISSUE THREAD

**Issue by Gcenx** [PhoenicisOrg/phoenicis-winebuild#64]
After the recent change it’s now possible to select proton and cx however this takes place after the listing pulled from wine’s git so selecting those means the build will error before it starts.

Instead wouldn’t it be better to ask before hand what your wanting to build then provide a list of options.

So selecting Proton would show proton versions same for WineCX (if you had others available that is)

~Or remove these two options from the menu and add examples for both that scripts that build both of these distributions.~

~I can add examples in my next pull-request,~ added WineCX examples to #67

**Comment by qparis** [comment]
Maybe we should let the user select any version?

**Comment by Gcenx** [comment]
> Maybe we should let the user select any version?

Yes when possible it should provide a version list after selecting from distributions.
But that depends on how many versions of WineCX you want to have available later, plus how would Proton be handled since each version head is in a different branch?

Or simple selecting cx or proton would just download head and build but selecting from "upstream, staging, dos_support" would then provide the wine version listing

**Comment by qparis** [comment]
Why not trusting the other and not listing anything?

**Comment by Gcenx** [comment]
> Why not trusting the other and not listing anything?

I'm sorry I don't follow what you mean here?

**Comment by qparis** [comment]
Our main problem is to find a suitable way to enumerate the versions for a given distribution. This enumeration is not required to trigger a build. You can use a raw input string instead. The drawback is that the user needs to know what version he can build and type it without mistake

**Comment by Gcenx** [comment]
> Our main problem is to find a suitable way to enumerate the versions for a given distribution. This enumeration is not required to trigger a build. You can use a raw input string instead. The drawback is that the user needs to know what version he can build and type it without mistake

Ah now I get what you mean.
>Why not trusting the other and not listing anything?

That was my other suggestion only list versions for wine but just get head for proton/winecx that would not be ideal but at least it would work.

**Comment by qparis** [comment]
I don’t think winebuild can build head for the moment

**Comment by Gcenx** [comment]
> I don’t think winebuild can build head for the moment

Hum then for now wouldn't it be safer to remove both cx and proton options from interactive_builder.py? it does include an example for cx already a proton one could also be added with easy.

**Comment by qparis** [comment]
Not sure, proton and cx properly supports version

**Comment by qparis** [comment]
They do not support listing version, this is a bit different

**Comment by Gcenx** [comment]
@qparis that's why I'm saying then would it not be better to just remove these options from interactive_builder.py then?

**Comment by qparis** [comment]
I don’t understand how it would help

**Comment by Gcenx** [comment]
> I don’t understand how it would help

I'm suggesting the options be temporally be removed from "interactive_builder.py" until those options can be used without failing.

**Comment by qparis** [comment]
They can actually. The only thing we cannot do with these build is enumarating the list of available versions. However, such a list exists. Just, you cannot see it. Appart from that, these distribution work exactly likr upstream and setting
